packages/extension-ui/src/Popup/Signing/Extrinsic.tsx

```diff
@@ -66,26 +68,38 @@ function renderMethod (data: string, { args, method }: Decoded, t: TFunction): R
     <>
       <tr>
         <td className='label'>{t<string>('method')}</td>
-        <td className='data'>
-          <details>
-            <summary>{method.section}.{method.method}{
-              method.meta
-                ? `(${method.meta.args.map(({ name }) => name).join(', ')})`
-                : ''
-            }</summary>
-            <pre>{JSON.stringify(args, null, 2)}</pre>
-          </details>
-        </td>
+        <td className='data'>{method.section}.{method.method}</td>
       </tr>
       {method.meta && (
-        <tr>
-          <td className='label'>{t<string>('info')}</td>
-          <td className='data'>
-            <details>
-              <summary>{method.meta.documentation.map((d) => d.toString().trim()).join(' ')}</summary>
-            </details>
-          </td>
-        </tr>
+        <>
+          {console.log(JSON.stringify(args, null, 2))}
+          {method.meta.args.map((arg, index) => (
+            <tr key={index}>
+              <td className='label'>{arg.name}</td>
+              <td className='data'>
+                {
+                  ADDRESS_TYPE.includes(arg.type.toHuman())
+                    ? (
+                      <Address
+                        address={args[index].Id || args[index]}
```

Comment:
This is ugly.
`LookupSource` type is an object with the `Id` field containing the address, an `AccountId` type arg however has the address right away. What's your take on that, what's the right way to take those out?

Also TS is complaining:
> Element implicitly has an 'any' type because expression of type 'number' can't be used to index type 'string | number | true | AnyJson[] | { [index: string]: AnyJson; }'

Is casting to something like ` { [index: string]: AnyJson; }` the only way to go?

Reply:
LookupSource can be multiple things -

- a straight AccountId
- an IndicesLookup
- a MultiAddress

Reply:
So the above doesn't work at all, it only caters for 2 of those (not correctly) since it actually should detect the type and then apply as required.

Reply:
Thanks, so do I understand correctly that the best way to do is to track down all the types that may have an address, and get the address out of each type accordingly? Is that even possible with `LookupSource` if you say what is behind may be some other type? How can I know what type is behind `LookupSource`?

Reply:
For instance in the case of MultiAddress (which is an enum) each entry behind it _is_ indeed a valid address. (So you cannot just check for Id)

The best way of getting the type and knowing how to handle it is actually via an instanceof check. Once this is in-place, you would handle each type. For instance on MultiAddress that will go live in Polkadot .28, the destination may be an AccountId or An AccountIndex, or a bare key.